### ISSUE TITLE
Crash when casting local size to short

### ISSUE THREAD

**Issue by jrprice** [google/clspv#555]
Test kernel:
```
kernel void foo(global int *out) {
  *out = (short)get_local_size(0);
}
```

Output:
```
$ bin/clspv foo.cl
@__spirv_WorkgroupSize = external local_unnamed_addr addrspace(8) global <3 x i32>Segmentation fault
```

The output when the issue is observed via clvk indicates stack corruption.

**Comment by alan-baker** [comment]
Another side effect of instcombine in llvm.

**Comment by alan-baker** [comment]
Fixed by #556
